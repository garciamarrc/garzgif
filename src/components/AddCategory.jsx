import { useState } from "react";

import { Button, Grid, TextField } from "@mui/material";
import SearchIcon from "@mui/icons-material/Search";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const cleanInput = inputValue.trim();
    if (cleanInput.length < 1) return;
    onNewCategory(cleanInput);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <Grid container>
        <Grid item xs={12}>
          <TextField
            sx={{ width: "100%" }}
            label="Buscar Gifs"
            value={inputValue}
            onChange={handleChange}
          />
        </Grid>
        <Grid item xs={12}>
          <Button
            sx={{ width: "100%" }}
            variant="contained"
            endIcon={<SearchIcon />}
            type="submit"
          >
            Buscar
          </Button>
        </Grid>
      </Grid>
    </form>
  );
};
