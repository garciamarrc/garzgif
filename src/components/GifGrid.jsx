import React from "react";

import { GifItem } from "./";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { LinearProgress } from "@mui/material";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <LinearProgress />}

      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
