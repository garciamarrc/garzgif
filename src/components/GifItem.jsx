import React from "react";

import PropTypes from "prop-types";

import { Card, CardContent, CardMedia, Typography } from "@mui/material";

export const GifItem = ({ title, url }) => {
  return (
    <Card sx={{ margin: 2, heigth: "auto" }}>
      <CardMedia component="img" image={url} alt={title} />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {title}
        </Typography>
      </CardContent>
    </Card>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
