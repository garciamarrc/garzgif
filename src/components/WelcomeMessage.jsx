import { Typography } from "@mui/material";
import { Box } from "@mui/system";
import React from "react";

export const WelcomeMessage = () => {
  return (
    <Box>
      <Typography variant="h3" sx={{ marginTop: 10 }}>
        Comienza a buscar <span style={{ color: "#F5A" }}>GIFS</span>
      </Typography>
    </Box>
  );
};
