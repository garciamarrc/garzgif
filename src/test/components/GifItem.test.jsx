import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe("Pruebas en <GifItem />", () => {
  const title = "Some";
  const url = "https://facebook.com/";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("El titulo debe renderizarse", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
