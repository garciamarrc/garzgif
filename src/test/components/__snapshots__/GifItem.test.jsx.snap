// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1f42516-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Some"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      src="https://facebook.com/"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Some
      </div>
    </div>
  </div>
</div>
`;
